proto/catalog.proto

```diff
@@ -247,6 +247,11 @@ message Table {
   // In older versions we can just initialize without it.
   bool cleaned_by_watermark = 30;
 
+  // This field is introduced in v1.2.0.
+  // It is only used if the state table is replicated.
+  // If replicated, only columns with index in `output_indices` will be read.
+  repeated int32 output_indices = 31;
+
```

Comment:
I'm thinking that `output_indices` should be a property of `ReplicatedStateTable`, rather than arrangement backfill itself.
Hence it is included as part of the protobuf definition.

Reply:
Should make it part of `ArrangementBackfill` first. We can still introduce breaking changes for it, we will mark arrangement backfill as experimental feature.